frontend/actions/src/main/java/com/virtuslab/gitmachete/frontend/actions/traverse/BaseTraverseAction.java

```diff
@@ -45,9 +47,40 @@ protected void onUpdate(AnActionEvent anActionEvent) {
     val repositorySnapshot = graphTable.getGitMacheteRepositorySnapshot();
     val branchLayout = repositorySnapshot != null ? repositorySnapshot.getBranchLayout() : null;
 
-    if (branchLayout == null || branchLayout.getRootEntries().isEmpty()) {
+    if (repositorySnapshot == null || branchLayout == null || branchLayout.getRootEntries().isEmpty()) {
       presentation.setEnabled(false);
       presentation.setDescription(getNonHtmlString("action.GitMachete.BaseTraverseAction.description.empty-layout"));
+      return;
+    }
+
+    String branchUnderAction = getNameOfBranchUnderAction(anActionEvent);
+    if (branchUnderAction == null) {
+      presentation.setEnabled(false);
+      presentation
+          .setDescription(getNonHtmlString("action.GitMachete.BaseTraverseAction.description.current-branch-unmanaged"));
+      return;
+    }
+
+    boolean anythingToBeDone = false;
+    for (var branch = branchUnderAction; branch != null; branch = branchLayout.findNextEntryName(branch)) {
```

Comment:
oh I see now. so it is disabled only when none of the next branches requires action - makes sense !